app/views/posts/_mdhint.html.erb

```diff
@@ -1,8 +1,15 @@
+<%
+  # defaults
+  category ||= defined?(category) ? category : nil
+  %>
```

Comment:
This doesn't work how you think it does - this block means you need to pass a `category` value to the `_mdhint` view when it's rendered, it doesn't automatically pull the category from elsewhere.

Reply:
Not too difficult, though: `mdhint` is rendered as part of `shared/body_field` on line 57 of `_form`. You should just need to copy this block to both `body_field` and `mdhint`, then add a `category: category` parameter to the `render` calls for each of them - line 57 in `_form`, and line 25 in `body_field`.

Reply:
I've made these changes and tested both creating and editing.